modules/Classroom/components/MaterialFormModal/MaterialFormModal.tsx

```diff
@@ -31,16 +35,35 @@ const MaterialFormModal: React.FC<IMaterialFormModalProps> = ({ opened, onClose
   });
 
   const [isLoading, setIsLoading] = useState(false);
+  const [uploadResource] = useUploadResourceMutation();
 
-  const onSubmit = (data: TMaterialFormValues) => {
-    // TODO : Add API call here
+  const onSubmit = async (data: TMaterialFormValues) => {
     setIsLoading(true);
-    console.log(data);
-    setTimeout(() => {
-      setIsLoading(false);
+    try {
+      await uploadResource({
+        file: data.file,
+        classroomId,
+        title: data.title,
+        description: data.description,
+      }).unwrap();
+
+      showNotification({
+        title: "Success",
+        message: "Resource uploaded successfully!",
+        color: "blue",
+      });
+
       reset();
       onClose();
-    }, 1000);
+    } catch (error) {
+      showNotification({
+        title: "Upload failed",
+        message: (error as ApiError).data?.message,
```

Comment:
Isn't there a parseApiErrorMessage method to structure the error messages?

Reply:
Not sure, I have created this ApiError earlier in the project so I am using that